utils/display/display.go

```diff
@@ -1,11 +1,88 @@
 package display
 
-func truncateString(oldString string, l int) string {
-	str := oldString
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
 
-	if len(oldString) > l {
-		str = oldString[:l] + "..."
+	"github.com/alexeyco/simpletable"
+)
+
+type Details map[string]string
+
+func PrintTable(obj interface{}, details Details) string {
+	bytes, err := json.Marshal(obj)
+	if err != nil {
+		panic(err)
+	}
+	amorphous := map[string]interface{}{}
+	err = json.Unmarshal(bytes, &amorphous)
+	if err != nil {
+		panic(err)
+	}
+
+	mainTable := simpletable.New()
+	for row, field := range details {
+		mainTable.Body.Cells = append(mainTable.Body.Cells, []*simpletable.Cell{
+			{Align: simpletable.AlignRight, Text: row + ":"},
+			{Text: fmt.Sprintf("%v", amorphous[field])},
+		})
+	}
+	mainTable.SetStyle(simpletable.StyleCompact)
+
+	return mainTable.String()
+}
+
+func interfaceSlice(slice interface{}) []interface{} {
+	s := reflect.ValueOf(slice)
+	if s.Kind() != reflect.Slice {
+		panic("InterfaceSlice() given a non-slice type")
+	}
+
+	// Keep the distinction between nil and empty slice input
+	if s.IsNil() {
```

Comment:
I'm not sure you'd ever hit this case. If the parameter `slice` is `nil` the conditional `if s.Kind() != reflect.Slice` evaluates to true.

Reply:
i made some changes and tested with with a hacked up unit test